icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/MeasureUnitTest.java

```diff
@@ -3340,65 +3340,248 @@ public void TestInvalidIdentifiers() {
 
                 // Compound units not supported in mixed units yet. TODO(CLDR-13700).
                 "kilonewton-meter-and-newton-meter",
-
         };
 
         for (String input : inputs) {
-            // TODO: test the failure for each input value.
-          MeasureUnit.forIdentifier(input);
+            try {
+                MeasureUnit.forIdentifier(input);
+                Assert.fail("An IllegalArgumentException must be thrown");
+            } catch (IllegalArgumentException e) {
+                continue;
+            }
         }
     }
 
     @Test
     public void TestCompoundUnitOperations() {
-        // TODO: implement
+        MeasureUnit.forIdentifier("kilometer-per-second-joule");
+
+        MeasureUnit kilometer = MeasureUnit.KILOMETER;
+        MeasureUnit cubicMeter = MeasureUnit.CUBIC_METER;
+        MeasureUnit meter = kilometer.withSIPrefix(MeasureUnit.SIPrefix.ONE);
+        MeasureUnit centimeter1 = kilometer.withSIPrefix(MeasureUnit.SIPrefix.CENTI);
+        MeasureUnit centimeter2 = meter.withSIPrefix(MeasureUnit.SIPrefix.CENTI);
+        MeasureUnit cubicDecimeter = cubicMeter.withSIPrefix(MeasureUnit.SIPrefix.DECI);
+
+        verifySingleUnit(kilometer, MeasureUnit.SIPrefix.KILO, 1, "kilometer");
+        verifySingleUnit(meter, MeasureUnit.SIPrefix.ONE, 1, "meter");
+        verifySingleUnit(centimeter1, MeasureUnit.SIPrefix.CENTI, 1, "centimeter");
+        verifySingleUnit(centimeter2, MeasureUnit.SIPrefix.CENTI, 1, "centimeter");
+        verifySingleUnit(cubicDecimeter, MeasureUnit.SIPrefix.DECI, 3, "cubic-decimeter");
+
+        assertTrue("centimeter equality", centimeter1 == centimeter2);
```

Comment:
Here and elsewhere: you can use `assertEquals` (for primitives and strings) or `assertSame` (for `==` objects) instead of `assertTrue`.  I think you still need `assertTrue` if you are testing the `.equals()` function.

Reply:
here we need to use `.equals` because centimeter is `MeasureUnit` object. I just forgot it